x/mongo/driver/mnet/connection.go

```diff
@@ -0,0 +1,127 @@
+// Copyright (C) MongoDB, Inc. 2023-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package mnet
+
+import (
+	"context"
+	"io"
+
+	"go.mongodb.org/mongo-driver/mongo/address"
+	"go.mongodb.org/mongo-driver/mongo/description"
+)
+
+// WireMessageReader represents a Connection where server operations can be
+// read from.
+type WireMessageReader interface {
+	Read(ctx context.Context) ([]byte, error)
+}
+
+// WireMessageWriter represents a Connection where server operations can be
+// written to.
+type WireMessageWriter interface {
+	Write(ctx context.Context, wm []byte) error
+}
+
+// WireMessageReadWriteCloser represents a Connection where server operations
+// can read from, written to, and closed.
+type WireMessageReadWriteCloser interface {
```

Comment:
Optional: Since `mnet` is dedicated to MongoDB connections, the `WireMessage` in `WireMessageReadWriteCloser` is redundant. Consider condensing the type name to `ReadWriteCloser`.